File: Android/scenes/convoai/src/main/java/io/agora/scene/convoai/ui/photo/PhotoResult.kt
package io.agora.scene.convoai.ui.photo

import android.graphics.Bitmap
import android.net.Uri
import java.io.File

/**
 * Photo processing result object
 * @param filePath Absolute file path, suitable for upload operations
 * @param fileUri File URI, suitable for Intent passing
 * @param file File object, suitable for file operations
 */
data class PhotoResult(
    val filePath: String,
    val fileUri: Uri,
    val file: File
) {
    val fileSize: Long get() = file.length()
    val fileName: String get() = file.name
    
    /**
     * Get file extension
     */
    fun getFileExtension(): String {
        return fileName.substringAfterLast('.', "")
    }
    
    /**
     * Get formatted file size
     */
    fun getFormattedFileSize(): String {
        val kb = fileSize / 1024.0
        val mb = kb / 1024.0
        
        return when {
            mb >= 1 -> String.format("%.2f MB", mb)
            kb >= 1 -> String.format("%.2f KB", kb)
            else -> "$fileSize B"
        }
    }
} 

File: iOS/Scenes/ConvoAI/ConvoAI/ConvoAI/Classes/Main/Photo/PhotoResult.swift
import UIKit
import Foundation

/**
 * Photo processing result object
 */
struct PhotoResult {
    let image: UIImage
    let filePath: String
    let fileURL: URL
    
    var fileName: String {
        return fileURL.lastPathComponent
    }
    
    var fileSize: UInt64 {
        do {
            let attributes = try FileManager.default.attributesOfItem(atPath: filePath)
            return attributes[.size] as? UInt64 ?? 0
        } catch {
            return 0
        }
    }
    
    /**
     * Get file extension
     */
    func getFileExtension() -> String {
        return fileURL.pathExtension
    }
    
    /**
     * Get formatted file size
     */
    func getFormattedFileSize() -> String {
        let kb = Double(fileSize) / 1024.0
        let mb = kb / 1024.0
        
        if mb >= 1 {
            return String(format: "%.2f MB", mb)
        } else if kb >= 1 {
            return String(format: "%.2f KB", kb)
        } else {
            return "\(fileSize) B"
        }
    }
} 